pkg/apis/authorization/types.go

```diff
@@ -17,13 +17,19 @@ limitations under the License.
 package authorization
 
 import (
+	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/unversioned"
 )
 
+// +genclient=true
+// +nonNamespaced=true
+// +noMethods=true
+
 // SubjectAccessReview checks whether or not a user or group can perform an action.  Not filling in a
 // spec.namespace means "in all namespaces".
 type SubjectAccessReview struct {
 	unversioned.TypeMeta
+	api.ObjectMeta
```

Comment:
David talked me around to the idea that many things that aren't classic
"objects" on the system have both name and namespace, resource version and
self link.  Some of those might use annotations for extensions (i.e. submit
a subject access review with an annotation for a custom feature not yet
implemented in the platform, or return an experimental feature in an
annotation).

If we have all of those, then what is left is basically object meta.  We
should definitely require those fields be empty for the short term, or
create a new restricted objectmeta if we do.

On Fri, Aug 5, 2016 at 1:54 AM, Eric Tune <notifications@github.com> wrote:

> In pkg/apis/authorization/types.go
> <https://github.com/kubernetes/kubernetes/pull/20573#discussion_r73645007>
> :
>
> >  // SubjectAccessReview checks whether or not a user or group can perform an action.  Not filling in a
> >  // spec.namespace means "in all namespaces".
> >  type SubjectAccessReview struct {
> >  	unversioned.TypeMeta
> > +	api.ObjectMeta
>
> I'm fine with validating that it is empty. I think that will allow us to
> back out the ObjectMeta once Clayton has a better fix.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/20573/files/42b8be284e0c00b2658d26d652153ea70c691626#r73645007>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p8zO8BsUFSyyqcGrd4iTVnlkPcfQks5qctAygaJpZM4HSwZB>
> .
>